stan/math/prim/err/check_simplex.hpp

```diff
@@ -35,29 +35,28 @@ namespace math {
  * @throw <code>std::domain_error</code> if the vector is not a
  *   simplex or if any element is <code>NaN</code>.
  */
-template <typename T, int R, int C>
-void check_simplex(const char* function, const char* name,
-                   const Eigen::Matrix<T, R, C>& theta) {
-  using size_type = index_type_t<Eigen::Matrix<T, Eigen::Dynamic, 1>>;
+template <typename T, require_eigen_t<T>* = nullptr>
+void check_simplex(const char* function, const char* name, const T& theta) {
   using std::fabs;
   check_nonzero_size(function, name, theta);
-  if (!(fabs(1.0 - theta.sum()) <= CONSTRAINT_TOLERANCE)) {
+  ref_type_t<T> theta_ref = theta;
```

Comment:
We do this because theta is used in two places? To prevent multiple evaluations?

Reply:
yes